docs/decisions/0003-learning-path-enrollment-api.rst

```diff
@@ -0,0 +1,125 @@
+
++---------------------+-------------------------------------------------------+
+| API Path            | /api/v1/learning-path-enrollment/                     |
++---------------------+-------------------------------------------------------+
+| Methods             | GET, POST, DELETE                                     |
++---------------------+-------------------------------------------------------+
+| Data                | learning_path_id (uuid), username (OPTIONAL)          |
++---------------------+-------------------------------------------------------+
+| Permissions Required| LoggedIn                                              |
++---------------------+-------------------------------------------------------+
+
+.. note:: The UUID of the learning_path_id will be replaced with a more
+   user-friendly value later.
+
+GET
+"""
+
+**Corresponding User Action:** Fetch my enrollment
+
+* The data fields ``learning_path_id`` and ``username`` will be passed as query
+  parameters.
+* Return the LearningPathEnrollment object if the logged-in user is enrolled in the
+  specified learning path. Otherwise returns a HTTP 404.
+* When the ``username`` is present, returns all the enrollments in a learning
+  path if the ``currentUser`` is a **staff** or **admin** user.
+
+
+POST
+""""
+
+**Corresponding User Action:** Enroll
+
+* When ``username`` is absent, enrolls the ``currentUser`` in the Learning
+  Path reference by the ``learning_path_id``.
+* When ``username`` is passed and is different from the ``currentUser``, the user
+  linked to ``username`` will be enrolled, if the ``currentUser`` is a **staff**
+  or **admin**.
+* For non-admin or non-staff users, when the ``currentUser`` and the user linked
+  to ``username`` are different, return a HTTP 403.
+
+DELETE
+""""""
+
+**Corresponding User Action:** Unenroll
+
+This will follow the same permission logic as the **POST** request allowing
+both self-unenrollment and staff-unenrollment.
+
+
+2. LearningPathEnrollment Model
+===============================
+
+Add a new ``enrolled_by`` as a ``ForeignKey`` field to the
+``LearningPathEnrollment`` model that will store the user who created the
+enrollment. This will allow for distinguishing between self and staff
+enrollments.
+
+Since it is already a subclass of the ``TimeStatmpedModel``, we don't need to
+introduce a ``enrolled_at`` field and can instead depend on the inherited ``created``
+field.
+
+Consequences
+************
+
+* The new API will allow external services with valid user credentials (eg.,
+  MFEs) to enroll learners in Learning Paths.
+* While access to courses can be restricted by marking them "Invitation
+  Required", there is no such flag for a Learning Path. So, any learner who
+  knows the ID of a Learning Path can enroll using the :ref:`Enroll API <enroll-api>`,
+  without actaully having access to any of the courses in the Learning Path. This
+  shortcoming will have be addressed in future development.
```

Comment:
Is this (i.e., ignoring the "invitation-only" setting) a bug or a feature? If this is not something we want, could you please expand this section to explain how we will address it?